From d4f3674bf5fb72ea8875a0cf06c37810eef0bc40 Mon Sep 17 00:00:00 2001
From: Maarten Arnst <maarten.arnst@uliege.be>
Date: Thu, 11 Jun 2026 08:46:49 +0200
Subject: [PATCH] clang cuda unreachable patch

---
 core/src/Kokkos_Macros.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/src/Kokkos_Macros.hpp b/core/src/Kokkos_Macros.hpp
index 834bb078e..98877c1bb 100644
--- a/core/src/Kokkos_Macros.hpp
+++ b/core/src/Kokkos_Macros.hpp
@@ -396,7 +396,7 @@
 // Define macro for unreachable code:
 // Only available in C++23
 // FIXME_HIP doesn't support std::unreachable in device code
-#if defined(__cpp_lib_unreachable) && !defined(KOKKOS_ENABLE_HIP)
+#if defined(__cpp_lib_unreachable) && !defined(KOKKOS_ENABLE_HIP) && !(defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_CLANG))
 #include <utility>
 #define KOKKOS_IMPL_UNREACHABLE() std::unreachable()
 #elif defined(__has_builtin)
